### PR TITLE
Poll logged in state

### LINKED ISSUE #186: Show logged-out users that they are logged out, even if mid-session
**Problem**
1. We currently automatically log you out ~after the initial time expires~ without clearly knowing why. This can even happen mid-session. The console shows 401 errors, but nothing in the editor UI prompts you to log in
2. This also affects tabs left open / computer went to sleep over weekend scenarios.

**Suggested solution**
1. Ideally, we don't log you out mid-session
2. If we do, we use the black bar with a link to re-authenticate, and the text "You were logged out. Click here to log in again".  Make sure the link opens in a new tab, so the work isn't lost.
3. (Make sure that new users _don't_ get that message, at least not in the first session)

### PR DESCRIPTION
**Pull request by balazsbajorics**
Fixes #186

image

**Problem:**
The editor would not tell users if they lost their login for any reason.

**Fix:**
Poll login status (I set it to every 5 seconds), and if the login status changes, dispatch an action. Show a toast. 
I also made a new kind of black status bar for users that were loggin in but are not logged in anymore. This new status bar opens the login screen in a new tab to avoid data loss.

**Commit Details:**
- Login status polling function
- I also wrapped the login status fetch to a try-catch to avoid an unhandled promise rejection error
- new `LOGIN_LOST` UserState to indicate a user that was logged in but not anymore
- a new `getLoginStateFromResponse` which sometimes returns the real login state coming from the server, or LoginLost (because LoginLost is not coming from the server, that is a UserState internal to the editor)
- I also noticed that if the server/internet is down, the black background png of the status bar can never load, so I changed it to have a backgroundColor and backgroundImage as well

### CODE REVIEW

File: editor/src/components/editor/notification-bar.tsx [hunk 1]
```diff
@@ -26,21 +26,38 @@ export const EditorOfflineBar = betterReactMemo('EditorOfflineBar', () => {
...
+          message={'Welcome to Utopia. Click here to sign in and save your projects.'}
+          onClick={onClickLoginSameTab}
+          style={{ cursor: 'pointer' }}
+        />
+      )
+    case 'LOGIN_LOST':
+      return (
+        <NotificationBar
+          level='ERROR'
+          message={'You were logged out. Click here to log in again.'}
```
Comment (maltenuhn) [inline editor/src/components/editor/notification-bar.tsx:53]:
```suggestion
          message={'You have been logged out. Click here to log in again and save your changes.'}
```

File: editor/src/components/editor/server.ts [hunk 2]
```diff
@@ -386,3 +392,20 @@ export async function downloadGithubRepo(
...
+  dispatch: EditorDispatch,
+  initialLoginState: LoginState,
+): void {
+  let previousLoginState: LoginState = initialLoginState
+  setInterval(async () => {
+    const loginState = await getLoginState('no-cache')
+    if (previousLoginState.type !== loginState.type) {
+      dispatch([setLoginState(loginState)])
+      if (isLoginLost(loginState)) {
+        dispatch([showToast(notice(`You have been logged out.`, 'ERROR', true))])
```
Comment (maltenuhn) [inline editor/src/components/editor/server.ts:406]:
```suggestion
        dispatch([showToast(notice(`You have been logged out. You can continue working, but your work won't be saved until you log in again.`, 'ERROR', true))])
```

File: editor/src/uuiui/styles/theme.ts [hunk 3]
```diff
@@ -522,14 +522,41 @@ const backgroundURLs = {
...
 
-const noticeStyles = {
-  success: { background: backgroundURLs.green, color: 'white' },
-  info: { background: '#f1f1f1', color: colorTheme.darkPrimary.value },
-  primary: { background: backgroundURLs.blue, color: 'white' },
-  notice: { background: backgroundURLs.paleblue, color: 'white' },
-  warning: { background: backgroundURLs.red, color: 'white' },
-  error: { background: backgroundURLs.almostBlack, color: 'white' },
-  disconnected: { background: backgroundURLs.noise, color: 'white' },
+const noticeStyles: { [styleName: string]: React.CSSProperties } = {
```
Comment (maltenuhn) [inline editor/src/uuiui/styles/theme.ts:525]: Thanks for spotting this. Classic mistake!

**Review by seanparsons — APPROVED** [review]
Is there a sensible way we can unit test this flow?

**Comment by balazsbajorics** [comment]
> Is there a sensible way we can unit test this flow?

hmm I guess I could write unit tests for the logic itself. the polling, not so much, I don't think so

**Comment by Rheeseyb** [comment]
When this happens (either via logging out in another tab or just straight up deleting the session token cookie), do you still see repeated server request failures with a 401 status?

**Review by Rheeseyb — APPROVED** [review]

**Review by enidemi — APPROVED** [review]

**Pull request merged**